### DISCUSSION TITLE
Could we seek to the begin of the stream in DiskWrite(string, long, Stream, CancellationToken)?

### DISCUSSION THREAD

**Discussion by fhebel** [jchristn/Blobject#4] [Ideas]
Ciao ragazzi,

I have a case where I can't change the code and my stream is at the end position.

Would you mind seeking to the begin of the stream, e.g.
```csharp
if (stream.CanSeek && stream.Length == stream.Position)
{ 
    stream.Seek(0, System.IO.SeekOrigin.Begin); 
}
```
in the method
```csharp
private async Task DiskWrite(string key, long contentLength, Stream stream, CancellationToken token)
```
?

Thanks a lot in advance

Best regards,

Florian

**Comment by fhebel** [comment]
Ciao ragazzi,

I solved my problem in another way. Nevertheless I think my proposal would be an improvement.

Tanti saluti

Florian

**Comment by jchristn** [comment]
Hi @fhebel thanks for your note.  I'd love to make this change, the challenge would be that it would break certain implementations.  For instance, one of the library's users has a ```FileStream``` where they want to extract a certain amount of data from a certain position, and they pass the ```FileStream``` positioned at the start of the range they need into ```public Task Write(string key, string contentType, long contentLength, Stream stream, CancellationToken token = default)```.  So if they need 1KB of data from a starting position of 16MB, the proposed change would break it.

**Reply by fhebel** [reply]
Hi @jchristn,

thanks for your fast reply.

>  one of the library's users has a FileStream where they want to extract a certain amount of data from a certain position, and they pass the FileStream positioned at the start of the range they need into public Task Write(string key, string contentType, long contentLength, Stream stream, CancellationToken token = default)

For that reason I proposed this change only for the very limited case that the position of the stream is equal to its length, which means that nothing (0 bytes) is going to be written otherwise ...

Tanti saluti

Florian

**Reply by jchristn** [reply]
Oh, got it.  I see that now, in that case you're 100% right it would be safe.  I'll make the change!

**Reply by jchristn** [reply]
I'm adding it here instead of within DiskWrite to catch it for all subordinate storage platforms.  
```public Task Write(string key, string contentType, long contentLength, Stream stream, CancellationToken token = default)```

**Reply by jchristn** [reply]
Here you go!

NuGet: https://www.nuget.org/packages/BlobHelper/2.1.4.4
Source: https://github.com/jchristn/BlobHelper/commit/e416ec94359b801818924acffdc10fc9075aaa11

**Reply by fhebel** [reply]
Thanks a lot!
